### ISSUE TITLE
Unexpected nullable warning in object initializer member using extension Add method

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#81102]
**Version Used**: 

**Steps to Reproduce**: [.NET Lab](https://lab.razor.fyi/#fVHRihMxFGVhVzRPsl9wwIdpX2bFBxFrC6UUWVEQt0-KaDpzZwxk7pTkZrelzB_4Lz75Bz4K_onfIOlk266IeUhI7j3n5J6jfpwp9da1tdNNXvjz72ePOFirl5ZAHA8VvOEaVxsv1Izu3PJZay0VYlr2-Uticqb4T8dl0wSJlCOlCqu9RxJWWwUAXrSYAtetKfFGGx4Md899Ma7LOYeGXKR44cUZricwQo3HGB8-jvZ9nzAG0w1mieEuy44pobYJ3-HiAoFpvaJCqMSNdhynmF09e_rk8XNMXR0aYkFbQTYrQvaPv2QoNHMrWBKCpxJV67DSTjck5JBZ4yU7MLw2Xg5Qw8jma8lpLcTetHxbGhz3DTOUgSAtSlNV5IgL8hErXwh9cMYa2UQVR6kBUc_newO63qlOdbc5zFICq7C0psCxYrJqi5pkBB-3LtndKZUi61nma0k8hyEW6f-LCeL4fyeaBHeRT8tycOzqYoJrbQP5IcY9Op-W5TvNNQ1SYT_I-WmjDb-6R861zr9_cP_3t19ff1YPTz-frE_-AA)

```cs
#nullable enable
using System;
using System.Collections.Generic;
using System.Collections.Immutable;

class Program
{
    static void Main()
    {
        IEnumerable<string> items = [];
        _ = new C()
        {
            Items = { items } // unexpected warning CS8620: Argument of type 'IEnumerable<string>' cannot be used for parameter 'list' of type 'List<string>' in 'Ext.extension<string>(List<string>)' due to differences in the nullability of reference types.
        };
    }
}

class C
{
    public List<string> Items { get; set; } = [];
}

static class Ext
{
    extension<T>(List<T> list)
    {
        public void Add(IEnumerable<T> values) => list.AddRange(values);
    }
}
```

**Expected Behavior**: No warnings are reported

**Actual Behavior**: A nullable warning is reported for 'items'. Note that the parameter which is being referenced is also unexpected,

The bug is may be related to the adjustments which are made to extension calls in order to include the receiver argument. We would want to ensure that the scenario behaves similarly to an classic `public static void Add<T>(this List<T> list, IEnumerable<T> values) => ...;` extension method.

**Comment by manfred-brands** [comment]
I have similar error when using a `param object?[]` or `params ReadOnlySpan<object?>` parameter.
```
    internal static class InvariantStringExtensions
    {
        extension(string)
        {
            public static string FormatInvariant(string format, params object?[] args)
            {
                return string.Format(CultureInfo.InvariantCulture, format, args);
            }
        }
    }

    internal static class InvariantStringExtensionsUsage
    {
        public static string TestFormatInvariant(string text, double value)
        {
            return string.FormatInvariant("{0} is {1}!", text, value);
        }
    }
```

<img width="1002" height="171" alt="Image" src="https://github.com/user-attachments/assets/daf39ba8-9b8d-4502-940c-4b769522bae5" />

**Comment by RikkiGibson** [comment]
Looks like #80024.

**Comment by aradalvand** [comment]
@RikkiGibson But why hasn't your fix #80433 made it to the .NET SDK yet? When can we expect to get the fix?

**Comment by AdmiralSnyder** [comment]
i'm running into this problem, getting CS8620 for collection initalization syntax, still, in SDK 10.0.400 - maybe this issue has gone stale and needs to be moved to a newer milestone?
